public/src/utils/Camera.js

```diff
@@ -0,0 +1,31 @@
+export class Camera {
+  constructor(canvas) {
+    this.zoomAmount = 1;
+    this.offset = { x: 0, y: 0 };
+    this.canvas = canvas;
+    this.maxZoom = 1;
+    this.minZoom = 0.2;
+    this.scrollSensitivity = 0.001;
+  }
+
+  set = (delta) => {
+    const rawUpdatedValue = this.zoomAmount + delta * this.scrollSensitivity;
+    const constrained = Math.min(
```

Comment:
zoom is limited to `0.2-1x`, i.e. you can zoom out but not in (beyond native 1x size). this is to prevent pixelation because the initial canvas is just a rasterized png so there's no way to redraw it at arbitrary scales.